### ISSUE TITLE
Question about building

### ISSUE THREAD

**Issue by Miannbaoo** [Ciremun/freedom#108]
Hello, i was wondering on how to build this myself as ive got all the req to build the files but didnt understand all fully since im still really new to this, perhaps is there a step by step or any better guide on how to do this? i would really appreciated for any kind of help.. and thank you for your work!!

**Comment by Ciremun** [comment]
1. press two keys simultaneously `Windows Key + R`
2. type `cmd.exe`, press enter
3. cd (change directory) into project folder
4. type `"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvarsall.bat" x86` or some other path, depending on visual studio directory
5. type `cl nobuild.c`
6. type `nobuild.exe`
7. ???? PROFIT!!!!

nobuild.exe has flags:

rebuild updated translation units (cpp files) / build everything if object file doesn't exist (in release mode)
```
nobuild.exe
```
rebuild all (including header files)
```
nobuild.exe rebuild
```
build in debug mode (rebuild flag may be required, since release and debug .obj files are not compatible)
```
nobuild.exe rebuild debug
```

**Comment by Miannbaoo** [comment]
?!?!!?THANKS!!!!!!

**Comment by Ciremun** [comment]
also
```
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build
```
this folder can be added to windows PATH, so that `vcvarsall` is always available in any new cmd.exe, so you don't need to remember the path.

you can also open cmd.exe faster by replacing this path
![image](https://github.com/Ciremun/freedom/assets/38132413/6b8f5164-5187-476f-9ade-68c3ec908a76)
with `cmd.exe` and pressing enter, it automatically opens a cmd.exe with changed directory, and you can just run `vcvarsall x86` once and then `nobuild.exe`, until the console is closed.

**Comment by Ciremun** [comment]
as a quality of life improvement, I think I could add automatic vcvarsall file detection, so that you don't even have to do anything with it

**Comment by Miannbaoo** [comment]
Noted! thanks alot for your help ^^/

**Comment by Miannbaoo** [comment]
Hi again! there seem to be bugs that caused to crash the game when Replay Copy are enabled for songs (in std mode, but all the other features are good without any issues) that are download manually from the beatmap lisiting site of osu!... meanwhile the rest of the functions of "freedom" are working fine for songs that are already inside the game when first installed including the Replay ><" hoping to know if im doing anything wrong in the process and is there any fix for it?

**Comment by Ciremun** [comment]
so. Replay Copy is enabled, right? At which point does it crash?

**Comment by Miannbaoo** [comment]
> so. Replay Copy is enabled, right? At which point does it crash?

yes! right as when i open up the preview page of player Replay then it crashed

**Comment by Ciremun** [comment]
i see, will be looking into it. the issue is that compressed_data_size is 0, and there is no check for that case

**Comment by Miannbaoo** [comment]
> i see, will be looking into it. the issue is that compressed_data_size is 0, and there is no check for that case

Ah..alright! thanks..

**Comment by Ciremun** [comment]
pushed a fix, didn't know score id takes 8 bytes xd

**Comment by Miannbaoo** [comment]
> pushed a fix, didn't know score id takes 8 bytes xd

thank you for the nice fix!
